src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -154,7 +154,25 @@ private Task<ApiHubFile> BuildFromAttribute(ApiHubFileAttribute attribute)
 
         private IFolderItem GetFileSource(string key)
         {
-            return _map[key];
+            if (_map.ContainsKey(key))
+            {
+                return _map[key];
+            }
+            else
+            {
+                // If the key doesn't exist, see if it is available as an environment variable.
+                // This might be the case for imperative binding scenarios.
+                var connectionString = Environment.GetEnvironmentVariable(key);
```

Comment:
>Environment.GetEnvironmentVariable [](start = 39, length = 34)

We should never call this directly and instead go through the INameResolver. (like we do in Script)